pkg/tsdb/prometheus/models/scope.go

```diff
@@ -0,0 +1,52 @@
+package models
+
+import (
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/promql/parser"
+)
+
+func ApplyQueryScope(rawExpr string, matchers []*labels.Matcher) (string, error) {
```

Comment:
I know it is experimental, but should we add some unit tests? At least for basic functionality?